include/openthread/platform/trel-udp6.h

```diff
@@ -0,0 +1,154 @@
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes the platform abstraction for Thread Radio Encapsulation Link (TREL) using an IPv6/UDP interface.
+ *
+ */
+
+#ifndef OPENTHREAD_PLATFORM_TREL_UDP6_H_
+#define OPENTHREAD_PLATFORM_TREL_UDP6_H_
+
+#include <stdint.h>
+
+#include <openthread/error.h>
+#include <openthread/instance.h>
+#include <openthread/ip6.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup plat-trel
+ *
+ * @brief
+ *   This module includes the platform abstraction for Thread Radio Encapsulation Link (TREL) using an IPv6/UDP
+ *   interface.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This function initializes the TREL IPv6/UDP interface.
+ *
+ * This function is called before any other TREL platform functions.
+ *
+ * @param[in] aInstance        The OpenThread instance structure.
+ * @param[in] aUnicastAddress  The unicast address to add to interface and use as tx source and rx destination.
+ * @param[in] aUdpPort         A UDP port number to use.
+ *
+ */
+void otPlatTrelUdp6Init(otInstance *aInstance, const otIp6Address *aUnicastAddress, uint16_t aUdpPort);
+
+/**
+ * This function updates the unicast IPv6 address for TREL IPv6/UDP interface.
+ *
+ * The interface should only have one unicast IPv6 address. Calling this function replaces any previously set unicast
+ * IPv6 address (during initialization from `otPlatTrelUdp6Init` or earlier calls to `otPlatTrelUdp6UpdateAddress()`).
+ *
+ * @param[in] aInstance        The OpenThread instance structure.
+ * @param[in] aUnicastAddress  The unicast address to add to interface and use for as tx source and rx destination.
+ *
+ */
+void otPlatTrelUdp6UpdateAddress(otInstance *aInstance, const otIp6Address *aUnicastAddress);
```

Comment:
Should we also allow the Udp Port to be updated in this API?

Reply:
The port number would be practically constant/same throughout device operation, so we set it from init function (no use-case/situation where we would need to change it afterwards).  The address however can potentially change during operation so we need a platform API to be able to change it.